configstack/running_module.go

```diff
@@ -184,13 +202,23 @@ func (module *runningModule) runNow() error {
 	}
 }
 
+var separator = strings.Repeat("-", 132)
+
 // Record that a module has finished executing and notify all of this module's dependencies
 func (module *runningModule) moduleFinished(moduleErr error) {
-	if moduleErr == nil {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished successfully!", module.Module.Path)
-	} else {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished with an error: %v", module.Module.Path, moduleErr)
+	status := "successfully!"
+	if moduleErr != nil {
+		status = fmt.Sprintf("with an error: %v", moduleErr)
+	}
+
+	module.Mutex.Lock()
+	defer module.Mutex.Unlock()
```

Comment:
Not the logging, it is thread safe.

But I got that kind of mixup on the output that is written directly to stdout and I got it with terragrunt without any of the recent change.

Just made tests and yes, I got that kind of mixup only on Windows...

But with the lock, there is no problem on Windows, nor on ios.